nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/java/org/apache/nifi/integration/versioned/ImportFlowIT.java

```diff
@@ -511,45 +511,45 @@ public void testUpdateFlowWithModifyingConnectionDeletingAndMovingPort() {
         final ProcessGroup groupA = createProcessGroup("group-a-id", "Group A", getRootGroup());
 
         //Create Process Group B under Process Group A
-        final ProcessGroup groupB = createProcessGroup("group-b-id", "Group B", groupA);
+        final ProcessGroup groupBunderA = createProcessGroup("group-b-id", "Group B", groupA);
 
         //Add Input port under Process Group B
-        final Port inputPort = getFlowController().getFlowManager().createLocalInputPort("input-port-id", "Input Port");
-        groupB.addInputPort(inputPort);
+        final Port inputPortBThenStayThenDelete = getFlowController().getFlowManager().createLocalInputPort("input-port-id", "Input Port");
+        groupBunderA.addInputPort(inputPortBThenStayThenDelete);
 
         //Add Processor 1 under Process Group A
-        final ProcessorNode processor1 = createProcessorNode(GenerateProcessor.class, groupA);
+        final ProcessorNode processorA1 = createProcessorNode(GenerateProcessor.class, groupA);
 
         //Add Processor 2 under Process Group A
-        final ProcessorNode processor2 = createProcessorNode(GenerateProcessor.class, groupA);
+        final ProcessorNode processorA2 = createProcessorNode(GenerateProcessor.class, groupA);
 
         //Add Output Port under Process Group A
-        final Port outputPort = getFlowController().getFlowManager().createLocalOutputPort("output-port-id", "Output Port");
-        groupA.addOutputPort(outputPort);
+        final Port outputPortAThenB = getFlowController().getFlowManager().createLocalOutputPort("output-port-id", "Output Port");
+        groupA.addOutputPort(outputPortAThenB);
 
         //Connect Processor 1 and Output Port as Connection 1
-        final Connection connection1 = connect(groupA, processor1, outputPort, processor1.getRelationships());
+        final Connection connectionProcessorA1ToOutputPortAThenProcessorA2 = connect(groupA, processorA1, outputPortAThenB, processorA1.getRelationships());
 
         //Connect Processor 1 and Input Port as Connection 2
-        final Connection connection2 = connect(groupA, processor1, inputPort, processor1.getRelationships());
+        final Connection connectionProcessorA1ToInputPortBThenStayThenDelete = connect(groupA, processorA1, inputPortBThenStayThenDelete, processorA1.getRelationships());
```

Comment:
Why the verbose naming of variables here? The variable name should be concise but description of what the object is. But not how it's going to be used. So I'm good with renaming to `connectionProcessorA1ToInputPortB` (though I personally don't think it's necessary, but I have no problem with it). But the `...ThenStayThenDelete` I do not think makes sense.

Reply:
I had hard time put together in my mind what was actually going on in this test. I had to rename the components just to be able to easily keep track of the flow while trying to understand the test case.

I think in general by far the best name for a variable is one that describes its purpose.
In tests the purpose is usually different than in production. Especially in this test their purpose is to play out a scenario. Hence the naming.

I understand that at first glance the names look verbose.
But when one wants to actually understand the multi-step test scenario, it can help tremendously.